Dalamud/Game/Gui/Nameplates/NameplateGui.cs

```diff
@@ -170,7 +170,7 @@ private nint HandleSetPlayerNameplateDetour(nint playerNameplateObjectPtr, bool
             iconId);
 
         // Free pointers
-        ptrToFree.ForEach(n => MemoryHelper.Free(n));
+        ptrToFree.ForEach(n => n.Dtor(true));
```

Comment:
You wrote Dtor(true), this will backfire. You didn’t let the game allocate the Utf8Strings, so it can’t free them. Call Dtor(false), so that it will free the StringPtr inside if needed (I say that because Utf8String has an internal buffer too) and let C#s garbage collector handle the Utf8Strings.